tests/results.cpp

```diff
@@ -1802,3 +1802,190 @@ TEST_CASE("results: snapshots") {
         CHECK_THROWS(snapshot.add_notification_callback([](CollectionChangeSet, std::exception_ptr) {}));
     }
 }
+
+TEST_CASE("distict")
+{
+    const int N = 10;
+    InMemoryTestFile config;
+    config.cache = false;
+    config.automatic_change_notifications = false;
+
+    auto r = Realm::get_shared_realm(config);
+    r->update_schema({
+        {"object", {
+            {"num1", PropertyType::Int},
+            {"string", PropertyType::String},
+            {"num2", PropertyType::Int}
+        }},
+    });
+
+    auto coordinator = _impl::RealmCoordinator::get_existing_coordinator(config.path);
+    auto table = r->read_group().get_table("class_object");
+
+    r->begin_transaction();
+    table->add_empty_row(N);
+    for (int i = 0; i < N; ++i) {
+        std::stringstream ss;
+        ss << "Foo_ " << i % 3;
+        table->set_int(0, i, i % 3);
+        table->set_string(1, i, StringData(ss.str().c_str()));
```

Comment:
`table->set_string(1, i, util::format("Foo_%1", i % 3).c_str());` for all of these.